### ISSUE TITLE
[recipe] Qwen3-Omni thinker separate-async recipe: colocated offload flags are pure overhead on the dedicated trainer pool and silently freeze training at parameter_sync_step>1 (pinned-verl snapshot aliasing)

### ISSUE THREAD

**Issue by zhtmike** [verl-project/verl-omni#645]
### System Info

- verl-omni: `main` @ `cf75c313`
- verl pin: `fefb080` (0.10.0.dev) — includes the affected helpers
- vllm-omni pin: `ded89346`
- Hardware concerned: single node, 4 × H800 80GB (2 FSDP trainer + 2 standalone
  rollout GPUs, the recipe's documented topology)
- Trainer mode: `trainer.v1.trainer_mode=omni_separate_async` (PR #380)

### Information

- [x] The official example scripts

### Tasks

- [x] An officially supported task in the `examples` folder

### Reproduction

Run the landed recipe as-is:

```bash
bash examples/gspo_trainer/qwen3_omni/run_qwen3_omni_thinker_gspo_lora_mmk12_separate_async_v1.sh
```

Relevant config (all shipped values, no overrides):

- `actor_rollout_ref.actor.fsdp_config.param_offload=true` (script `:66`)
- `actor_rollout_ref.actor.fsdp_config.optimizer_offload=true` (script `:67`)
- `trainer.v1.separate_async.parameter_sync_step=8` (script `:96`) with
  `data.train_batch_size=128` (`:37`) and `ppo_mini_batch_size=16` (`:56`) —
  8 local updates per global step
- `trainer.n_gpus_per_node=2` (`:103`) — the trainer FSDP2 pool is **dedicated**;
  `rollout.gpu_memory_utilization=0.8` governs only the 2 standalone rollout GPUs

**Problem 1 — pure overhead.** The offload flags exist for colocation, where the
actor and the vLLM engine share GPUs and the actor must vacate to CPU during
rollout. Under separate-async nothing else claims the trainer GPUs, so the flags
only buy 16+ full-parameter CPU↔GPU swaps of the sharded 30B weights per global
step (one `old_log_prob` + one `update_actor` phase per local update), folded
into `timing_s/old_log_prob` and `timing_s/update_actor`.

**Problem 2 — silent training freeze (parameter corruption).** With
`param_offload=true`, parameters are CPU-resident when the decoupled-PPO
snapshot dance runs. The pinned verl's `fsdp2_sharded_save_to_cpu`
(`verl/utils/fsdp_utils.py:1146-1185`) snapshots with
`tensor.detach().cpu()` — at both the non-DTensor branch (`:1167`) and the
DTensor shard branch (`:1180`) — which **returns a storage-sharing view** for
CPU-resident tensors. `DetachActorWorker` wires this helper in directly
(`verl/experimental/separation/engine_workers.py:96-99`), no copy in between.
The dance (`verl/trainer/ppo/v1/trainer_separate_async.py:154-185`, the
`local_trigger_step >= 1` branch) then corrupts the snapshot:

1. `save_model_to_cpu(n)` — slot *n* aliases the live storage holding the
   trained weights W_n.
2. `restore_model_from_cpu(0)` — the in-place `copy_` of the cycle-start weights
   W₀ into the live storage rewrites slot *n* through the alias; W_n is gone.
3. `restore_model_from_cpu(n)` — installs W₀. Every local update resets to the
   cycle-start policy.

This was reproduced end-to-end on the MiniCPM-o 4.5 port of this same recipe
(same trainer, same worker, same verl helpers, same flags): 13 global steps of
healthy-looking metrics with a policy pinned at init. To check whether any
existing Qwen3-Omni run at `parameter_sync_step > 1` was affected, read the
curves for the signature below — it requires no rerun.

**Why the #380 GPU validation did not catch it:** the smoke
(`tests/special_e2e/run_gspo_qwen3_omni_thinker_lora_v1_separate_async_smoke.sh`)
sets the same offload flags (`:81-82`) but runs `parameter_sync_step=1`
(`:114`). At `parameter_sync_step=1` every step takes the
`local_trigger_step == 0` branch — a single `save_model_to_cpu(0)` and a direct
compute, no restore dance — so the hazard is structurally unreachable there;
only the recipe's `parameter_sync_step=8` shape is exposed.

### Expected behavior

- The recipe should not carry the colocated offload flags: the trainer GPUs are
  dedicated, and with parameters GPU-resident at snapshot time `.cpu()` performs
  a real copy, so the aliasing hazard cannot trigger and the swap overhead
  disappears:

  ```diff
  -    actor_rollout_ref.actor.fsdp_config.param_offload=true \
  -    actor_rollout_ref.actor.fsdp_config.optimizer_offload=true \
  +    actor_rollout_ref.actor.fsdp_config.param_offload=false \
  +    actor_rollout_ref.actor.fsdp_config.optimizer_offload=false \
  ```

  Memory headroom on the 2-GPU trainer pool should be smoke-checked first: the
  30B bf16 weights shard to ~30 GB/rank, LoRA-only optimizer states are small,
  and gradient checkpointing is on — expected to fit 80 GB cards comfortably,
  but tighter than the colocated shape.

- Regardless of the recipe flip, snapshots should own their storage so any run
  that *must* keep offload (bigger models, fewer trainer GPUs) is safe:
  upstream `fsdp2_sharded_save_to_cpu` should copy
  (`.to("cpu", copy=True)` / `.clone()` instead of `detach().cpu()`), or
  verl-omni can wrap the save handler in
  `OmniDetachActorWorker._get_strategy_handlers` as defense-in-depth with a CPU
  test.

### Logs

No error is raised — that is the defect. Frozen-run metric signature (observed
on the MiniCPM-o sibling run of this recipe; identical mechanism and flags):

```
actor/ppo_kl          == 0.0   (bit-exact at every step: old == new log-probs)
actor/pg_clipfrac     == 0.0   (bit-exact at every step)
actor/pg_loss         ~ 1e-9
actor/grad_norm       ~ 0.17   (nonzero — updates run, they just reset each time)
training score        flat     (MiniCPM: 0.36–0.42; val@10 0.414 vs the
                               colocated baseline's train score 0.708 at step 10)
```

### Who can help?

@knlnguyen1802 (separate-async trainer, PR #380) @ZihaoW123 (rollout/trainer)

### References

- Recipe: `examples/gspo_trainer/qwen3_omni/run_qwen3_omni_thinker_gspo_lora_mmk12_separate_async_v1.sh` (PR #380)
- Smoke: `tests/special_e2e/run_gspo_qwen3_omni_thinker_lora_v1_separate_async_smoke.sh`
- Sibling occurrence, end-to-end reproduction + root cause: the MiniCPM-o 4.5
  separate-async recipe (PR #629), whose recipe already flips both flags to
  `false`
- Upstream: verl `fsdp2_sharded_save_to_cpu` view aliasing under CPU-resident
  params (to be filed against verl)
